pkg/helm/envclient_factory.go

```diff
@@ -0,0 +1,63 @@
+package helm
+
+import (
+	"context"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	configcli "github.com/aws/eks-anywhere/pkg/config"
+	"github.com/aws/eks-anywhere/pkg/registrymirror"
+)
+
+// EnvClientFactory provides a Helm client for a cluster.
+type EnvClientFactory struct {
+	helmClient RegistryClient
+	builder    ExecutableBuilder
+}
+
+// NewEnvClientFactory returns a new EnvClientFactory.
+func NewEnvClientFactory(builder ExecutableBuilder) *EnvClientFactory {
+	return &EnvClientFactory{
+		builder: builder,
+	}
+}
+
+// buildClient returns a new helm executable client.
+func (f *EnvClientFactory) buildClient(opts ...Opt) ExecuteableClient {
+	return f.builder.BuildHelmExecutable(opts...)
+}
+
+// Get returns the helm registry client.
+// The parameters here are not used and it only returns the client that is initialized using Init.
+func (f *EnvClientFactory) Get(_ context.Context, _ *anywherev1.Cluster) (RegistryClient, error) {
+	return f.helmClient, nil
+}
+
+// Init builds the helm registry client once using the registry mirror information from the cluster information.
+// It should be called at least once first, before trying to retrieving and using the client using Get.
+// It only builds the helm registry client once.
```

Comment:
Let's add a comment saying this is not thread safe and the caller should guarantee that it doesn't get called from multiple threads